Extensions/TileMap/collision/TransformedTileMap.ts

```diff
@@ -0,0 +1,747 @@
+/// <reference path="../helper/TileMapHelper.d.ts" />
+namespace gdjs {
+  export namespace TileMap {
+    /**
+     * A tile map transformed with an affine transformation.
+     *
+     * @see {@link getHitboxesAround} It gives a fast access to hitboxes for collision handling.
+     */
+    export class TransformedCollisionTileMap {
+      /**
+       * The model that describes the tile map.
+       */
+      private _source: TileMapHelper.EditableTileMap;
+      tag: string;
+      private _layers: Map<integer, TransformedCollisionTileMapLayer>;
+      // TODO Tiled allows to offset the layers
+      /**
+       * The transformation from the time map coordinate (in pixels)
+       * to the scene coordinate (in pixels).
+       */
+      private _transformation: gdjs.AffineTransformation = new gdjs.AffineTransformation();
+      /**
+       * The transformation from the scene coordinate (in pixels)
+       * to the time map coordinate (in pixels).
+       */
+      private _inverseTransformation: gdjs.AffineTransformation = new gdjs.AffineTransformation();
+      /**
+       * This allows tiles to know if their hitboxes must be updated.
+       * @see {@link TransformedCollisionTile.affineTransformationUpToDateCount}
+       */
+      _transformationUpToDateCount: integer = 1;
+      /**
+       * An reusable Point to avoid allocations.
+       */
+      private static readonly workingPoint: FloatPoint = [0, 0];
+
+      /**
+       * @param source The model that describes the tile map.
+       */
+      constructor(source: TileMapHelper.EditableTileMap, tag: string) {
+        this._source = source;
+        this.tag = tag;
+        this._layers = new Map<integer, TransformedCollisionTileMapLayer>();
+        for (const sourceLayer of source.getLayers()) {
+          // TODO should there be a list of EditableTileMapLayer to avoid a cast?
+          const tileLayer = sourceLayer as TileMapHelper.EditableTileMapLayer;
+          if (!tileLayer) {
+            // TODO Handle collision mask on object layers?.
```

Comment:
Don't formulate this as a question but rather as an affirmation: "This is skipped because we don't handle collisions on object layers."